### Leitura do arquivo .pkl

Leitura do arquivo onde foi salvo os previsores e as classes com seu devido pre processsamento

In [28]:
import pickle

In [29]:
with open('../../variaveis_preprocessadas/credit.pkl', 'rb') as f:
  previsores_treinamento, classe_treinamento, previsores_teste, classe_teste = pickle.load(f)

### Utilização do algoritmo Naive Bayes

In [30]:
from sklearn.naive_bayes import GaussianNB

gausian = GaussianNB()
gausian.fit(previsores_treinamento, classe_treinamento)

previsoes = gausian.predict(previsores_teste)

### Avaliação dos resultados do algoritmo Naive Bayes

In [31]:
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

In [32]:
score = accuracy_score(classe_teste, previsoes)
score

0.936

In [33]:
confusion_matrix(classe_teste, previsoes)

array([[423,   9],
       [ 23,  45]], dtype=int64)

In [35]:
print(classification_report(classe_teste, previsoes))

              precision    recall  f1-score   support

           0       0.95      0.98      0.96       432
           1       0.83      0.66      0.74        68

    accuracy                           0.94       500
   macro avg       0.89      0.82      0.85       500
weighted avg       0.93      0.94      0.93       500



### Resultados obtidos

O algoritmo possui 93% de acurácia. O algoritmo para a deteção de pessoas que não vão pagar o emprestimo consegue detectar 66% e possui uma acurácia de 83%. Como a atividade principal é a detecção das pessoas que não pagarão o emprestimo, com está abordagem possui um rendimento considerado como aceitável.